<launch>

    <include file="$(find quad_utils)/launch/load_params.launch" />

    <arg name="global_planner" default="twist" />
    <arg name="local_planner" default="full" />
    <arg name="logging" default="false" />
    <arg name="twist_input" default="none" />

    <!-- Launch the node to publish the global body plan -->
    <group if="$(eval arg('global_planner') == 'fgmp')">
        <node name="global_body_planner" pkg="global_body_planner" type="global_body_planner_node" output="screen" />
        <param name="local_planner/use_twist_input" value="false" />
    </group>

    <!-- Use twist input to local plan and no global plan -->
    <group if="$(eval arg('global_planner') == 'twist')">
        <param name="local_planner/use_twist_input" value="true" />

        <group if="$(eval arg('twist_input') == 'keyboard')">
            <node name="teleop_twist_keyboard" pkg="teleop_twist_keyboard" type="teleop_twist_keyboard.py" />
        </group>

        <group if="$(eval arg('twist_input') == 'joy')">
            <include file="$(find teleop_twist_joy)/launch/teleop.launch">
                <arg name="joy_config" value="ps3-holonomic" />
            </include>
        </group>
    </group>

    <node name="local_planner" pkg="local_planner" type="local_planner_node" output="screen" />

    <group if="$(arg logging)">
        <include file="$(find quad_utils)/launch/logging.launch" />
    </group>

</launch>